pkg/logs/tag/local_provider.go

```diff
@@ -39,18 +42,21 @@ func NewLocalProvider(t []string) Provider {
 
 			p.Lock()
 			defer p.Unlock()
-			p.expectedTags = p.tags
+			p.expectedTags = nil
 		}()
 	}
 
 	return p
 }
 
-// GetTags returns an empty list of tags.
+// GetTags returns the list of locally-configured tags.  This will include the
+// expected tags until the expected-tags deadline, if those are configured.
```

Comment:
What do you think about adding a comment that the return value `[]string` must not be mutated as several goroutines can manipulate it?

Reply:
Good idea :)